.codegen/__init__.py.tmpl

```diff
@@ -146,10 +146,17 @@ class AccountClient:
         :param workspace: The workspace to construct a client for.
         :return: A ``WorkspaceClient`` for the given workspace.
         """
-        config = self._config.copy()
-        config.host = config.environment.deployment_url(workspace.deployment_name)
-        config.azure_workspace_resource_id = azure.get_azure_resource_id(workspace)
-        config.account_id = None
+        config_inner = self._config.as_dict().copy()
+        if 'host' in config_inner:
+            del config_inner['host']
```

Comment:
using something like:
`keys_to_remove = ['host', 'azure_workspace_resource_id', 'account_id']`

might be better for readability / extensible in case we want to have more keys we want to remove in future.